common/clock/ratelimiter_test.go

```diff
@@ -0,0 +1,434 @@
+// The MIT License (MIT)
```

Comment:
a benchmark test would also be useful to see the impact of locking and whether it's even an issue for the per-second call volume we need to support.

Reply:
yea, I'm kinda curious about it too.  might try to build it today yet.

my _guess_ is that it won't be too much worse in practice, since `*rate.Limiter` has a mutex in it already... though it does clearly spend effort to keep the lock held as short as it can be inside each method, and this raises that window to cover at least the whole method.